test/integration/api-create.test.js

```diff
@@ -90,250 +93,481 @@ test('setup', function (tt) {
+//             dockerClient: DOCKER_ALICE,
+//             test: t,
+//             extra: { 'HostConfig.Memory': memory }
+//         }, oncreate);
+
+//         function oncreate(err, result) {
+//             t.ifErr(err, 'create container');
+//             created = result;
+//             t.equal(created.vm.ram, MEMORY_IN_MBS,
+//                     'VM should be created with ' + MEMORY_IN_MBS
+//                     + 'MBs of RAM');
+//             t.end();
+//         }
+//     });
+
+//     tt.test('docker rm', function (t) {
+//         DOCKER_ALICE.del('/containers/' + created.id, ondel);
+
+//         function ondel(err, res, req, body) {
+//             t.ifErr(err, 'rm container');
+//             t.end();
+//         }
+//     });
+
+//     tt.test('docker create with invalid memory',
+//             function (t) {
+//         h.createDockerContainer({
+//             vmapiClient: VMAPI,
+//             dockerClient: DOCKER_ALICE,
+//             test: t,
+//             extra: { 'HostConfig.Memory': 'Foo' }
+//         }, oncreate);
+
+//         function oncreate(err, result) {
+//             t.ifErr(err, 'create container');
+//             created = result;
+//             t.equal(created.vm.ram, CONFIG.defaultMemory,
+//                     'VM should be created with ' + CONFIG.defaultMemory
+//                     + 'MBs of RAM');
+//             t.end();
+//         }
+//     });
+
+//     tt.test('docker rm', function (t) {
+//         DOCKER_ALICE.del('/containers/' + created.id, ondel);
+
+//         function ondel(err, res, req, body) {
+//             t.ifErr(err, 'rm container');
+//             t.end();
+//         }
+//     });
+// });
+
+
+// test('api: create with env var that has no value (DOCKER-741)', function (tt) {
+//     tt.test('create empty-env-var container', function (t) {
+//         h.createDockerContainer({
+//             vmapiClient: VMAPI,
+//             dockerClient: DOCKER_ALICE,
+//             test: t,
+//             extra: { 'Env': ['ENV_NO_VALUE'] },
+//             start: true  // Will start the container after creating.
+//         }, oncreate);
+
+//         function oncreate(err, result) {
+//             t.ifErr(err, 'create empty-env-var container');
+//             t.equal(result.vm.state, 'running', 'Check container running');
+//             DOCKER_ALICE.del('/containers/' + result.id + '?force=1', ondelete);
+//         }
+
+//         function ondelete(err) {
+//             t.ifErr(err, 'delete empty-env-var container');
+//             t.end();
+//         }
+//     });
+// });
+
+test('ensure fabrics enabled', function (tt) {
+    tt.test('fabric configuration', function (t) {
+        var listOpts = {};
+        var listParams = {};
+        NAPI.listFabricVLANs(ALICE.account.uuid, listOpts, listParams,
+            function (err, vlans) {
+                if (err) {
+                    FABRICS = false;
+                    if (err.code !== 'PreconditionRequiredError') {
```

Comment:
What does `'PreconditionRequiredError'` represent? Are fabrics enabled in nightly or other DCs where tests are run automatically?